src/lib/route-broadcaster.js

```diff
@@ -133,95 +134,66 @@ class RouteBroadcaster {
   }
 
   _broadcastToLedger (adjacentLedger, routes, unreachableLedgers, requestFullTable) {
-    const connectors = Object.keys(this.peersByLedger[adjacentLedger])
-    return Promise.all(connectors.map((account) => {
-      log.info('broadcasting ' + routes.length + ' routes to ' + account)
-      const routesNewToConnector = routes.filter((route) => (route.added_during_epoch > (this.peerEpochs[account] || -1)))
-      const fieldsToOmit = ['added_during_epoch']
-      if (!this.broadcastCurves) {
-        fieldsToOmit.push('points')
-      }
+    log.info('broadcasting ' + routes.length + ' routes to ' + adjacentLedger)
+    const routesNewToConnector = routes.filter((route) => (route.added_during_epoch > (this.peerEpochs[adjacentLedger] || -1)))
+    const fieldsToOmit = ['added_during_epoch']
+    if (!this.broadcastCurves) {
+      fieldsToOmit.push('points')
+    }
 
-      const newRoutes = routesNewToConnector.map((route) => _.omit(route, fieldsToOmit))
-      if (unreachableLedgers.length > 0) log.info('_broadcastToLedger unreachableLedgers:', unreachableLedgers)
+    const newRoutes = routesNewToConnector.map((route) => _.omit(route, fieldsToOmit))
+    if (unreachableLedgers.length > 0) log.info('_broadcastToLedger unreachableLedgers:', unreachableLedgers)
 
-      const broadcastPromise = this.ledgers.getPlugin(adjacentLedger).sendRequest({
-        ledger: adjacentLedger,
-        from: this.ledgers.getPlugin(adjacentLedger).getAccount(),
-        to: account,
-        custom: {
-          method: 'broadcast_routes',
-          data: {
-            new_routes: newRoutes,
-            hold_down_time: this.holdDownTime,
-            unreachable_through_me: unreachableLedgers,
-            request_full_table: requestFullTable
-          }
-        },
-        // timeout the plugin.sendRequest Promise just so we don't have it hanging around forever
-        timeout: this.routeBroadcastInterval
-      })
+    const broadcastPromise = this.ledgers.getPlugin(adjacentLedger).sendRequest({
```

Comment:
Does `this.ledgers.getPlugin(adjacentLedger)` return an LPI1 or an LPI2 interface? Because the `sendRequest` method, needed for broadcasting routes, seems to be [absent in LPI2](https://github.com/interledger/rfcs/blob/8ca2801402fa84f52b4399047911055c6200a09f/0000-ledger-plugin-interface-2/0000-ledger-plugin-interface-2.md#methods)?

Reply:
It's not in the spec but it is in the [`ilp-compat-plugin`](https://github.com/interledgerjs/ilp-compat-plugin/blob/master/index.js#L125-L142) as a deprecated feature

Reply:
`sendRequest` won't exist when `compat()` returns the original LPI2 plugin (https://github.com/interledgerjs/ilp-compat-plugin/blob/d8331126a92ad2b65a7e7c4513ce44c13d485340/index.js#L31-L33).

Reply:
So then this PR will break route broadcasts?

Reply:
LPI1 plugins will work thanks to `compat`; LPI2 plugins will be broken.

Reply:
OK, but is there a plan to fix them, then? Or are we deprecating route broadcasts?

Reply:
They should be switched to use either interledger payments or the internet

Reply:
in a similar way to @justmoon 's DHCP on ILP proposal (for asking your peer to assign you an ILP address), it would be easy to send a payment to your peer for which the destination is something like `peer.broadcast`, and the condition is either trivially fulfillable (if you're paying to send broadcasts) or unfulfillable (if you're not paying to send broadcasts). I don't see why that's worse/harder than using messaging.

Reply:
Sure, that's what plugin-xrp-escrow [already does](https://github.com/ripple/ilp-plugin-xrp-escrow/blob/master/src/plugin.js#L385). And if a BTP connection is available between the connectors then they can just use a BTP Message packet. But that's what `plugin.sendRequest` already nicely abstracts, and we agree that ilp-connector needs it, so then instead of removing code from the stack, we have just split the ledger plugin into a separate transfer plugin and messaging plugin, which is just moving code around.

Reply:
We don't actually need a messaging plugin anymore though, because transfers are expected to be just as fast as messages. And setting the amount to 1 means that the impact on your liquidity is negligible. This just lets us keep plugins as simple as possible, which should make them less buggy and more secure

Reply:
Ben and I discussed this on Slack and reduced this question to https://github.com/interledger/rfcs/issues/357